app/lib/bk/compat/parsers/bitrise/workflows.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+module BK
+  module Compat
+    # Bitrise translation of workflows
+    class Bitrise
+      private
+
+      def parse_workflow(wf_name, _wf_config)
+        raise "No key within loaded workflow configuration for #{wf_name}" unless @steps_by_key.include?(wf_name)
+
+        @steps_by_key[wf_name]
```

Comment:
is there a reason to have this indirection? why not just parse the workflows directly into the step instead of loading and then just referencing them?

Reply:
The main thinking behind the `loader` is to effectively load in each `workflow`s' base configuration only (key point) to their specific steps (right now its just `CommandStep`s) - effectively its main responsibility as of now is loading the  commands by way of each step (and what is currently translatable)

It'll be the job of `parse_workflow` (and more future `def`s) that'll modify the base step through with additional properties (more example, dependencies etc) and tweak them before the parser renders out the configured pipeline. Effectively its 'similar' in a way to how Circle's parser does it through `@commands_by_key` - foundation layer (loader) and then adding bits/pieces to the config on-top

Reply:
I understand the similarities with CircleCI, but that was done that way because CircleCI's YAML has user-defined commands and jobs that can be instantiated in workflows (or other jobs/commands) so there was a clear dependency and reason for the loading. In this case, like any other parses that is not CircleCI, there is no such flexibility so I don't think that this is necessary

Reply:
Thinking more - I'm probably thinking two steps ahead from whats needed really - I reverted this back to the more simpler mechanism and not referencing an instance hash to pick out the steps per workflow.

For other more complex parsers - it might make sense, but its more simpler for now here 👍